src/components/Board/BoardShare/BoardShare.component.js

```diff
@@ -67,28 +75,36 @@ const BoardShare = ({
       </DialogTitle>
       <DialogContent className="ShareDialog__content">
         <div className="ShareDialog__Subtitle">
-          <div>{isOwnBoard && <Button
-            color="primary"
-            variant={isPublic ? "outlined" : "contained"}
-            className="ShareDialog__ToggleStatusButton"
-            onClick={publishBoard}
-            >
-              {!isPublic? <FormattedMessage {...messages.publishBoard } /> : <FormattedMessage {...messages.unpublishBoard } />}
-            </Button>
+          <div>{
+            isOwnBoard  
+              ? <Button
+              color="primary"
+              variant={isPublic ? "outlined" : "contained"}
+              className="ShareDialog__ToggleStatusButton"
+              onClick={publishBoard}
+              >
+                {!isPublic? <FormattedMessage {...messages.publishBoard } /> : <FormattedMessage {...messages.unpublishBoard } />}
+              </Button>
+              : <><Alert severity="warning">warning — you need to be registered to share a board</Alert>
```

Comment:
This should use the `unregisteredWarning` message.

Reply:
i mean, i had confusion related to branches because this is about another issue and branch.